client/ayon_applications/manager.py

```diff
@@ -537,27 +933,85 @@ def _run_process(self):
                 if key in app_env
             }
 
-        # Create temp file
+        # Create the temp file
         json_temp = tempfile.NamedTemporaryFile(
-            mode="w", prefix="op_app_args", suffix=".json", delete=False
+            mode="w", prefix="ay_app_args", suffix=".json", delete=False
         )
         json_temp.close()
-        json_temp_filpath = json_temp.name
-        with open(json_temp_filpath, "w") as stream:
+        json_temp_filepath = json_temp.name
+        with open(json_temp_filepath, "w") as stream:
             json.dump(json_data, stream)
 
-        launch_args.append(json_temp_filpath)
+        launch_args.append(json_temp_filepath)
 
         # Create mid-process which will launch application
         process = subprocess.Popen(launch_args, **self.kwargs)
         # Wait until the process finishes
         #   - This is important! The process would stay in "open" state.
         process.wait()
+
+        # read back pid from the json file
+        try:
+            with open(json_temp_filepath, "r") as stream:
+                json_data = json.load(stream)
+
+                process_info = ProcessInfo(
+                    name=self.application.full_name,
+                    args=self.launch_args,
+                    env=self.kwargs.get("env", {}),
+                    cwd=os.getcwd(),
+                    pid=json_data.get("pid"),
+                    output=Path(temp_file.name),
+                    site_id=get_local_site_id()
+                )
+                # Store process info to the database
+                self.manager.store_process_info(process_info)
+        except OSError as e:
+            self.log.error(
+                "Failed to read process info from JSON file: %s", e,
+                exc_info=True
+            )
+
         # Remove the temp file
-        os.remove(json_temp_filpath)
+        os.remove(json_temp_filepath)
         # Return process which is already terminated
         return process
 
+    def _execute_with_stdout(self):
+        """Run the process with stdout and stderr redirected to a file.
+
+        Stores process information to the database.
+
+        Returns:
+            subprocess.Popen: The process object created by Popen.
+        """
+        temp_file = tempfile.NamedTemporaryFile(
+            mode="w",
+            prefix=f"ayon_{self.application.host_name}_output_",
+            suffix=".txt",
+            delete=False
+        )
+        temp_file_path = temp_file.name
+        temp_file.close()
+        with open(temp_file_path, "wb") as tmp_file:
+            self.kwargs["stdout"] = tmp_file
+            self.kwargs["stderr"] = tmp_file
+            process = subprocess.Popen(self.launch_args, **self.kwargs)
```

Comment:
This won't create detached process on windows.

Reply:
If you launch Nuke or Photoshop, it is separated and the stdout and stderr are not outputed to AYON terminal?